src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,10 +60,9 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request@Publish(_, _, _, _, _, _) => {
```

Comment:
this description of typed patterns is actually very vague. a typed pattern is a type cast, and is not pure. using the constructor pattern is generally the correct solution, and yes, the only way to instead of deconstructing the case class, but referring to the whole thing, is using the @ binder.